src/jdk.jdi/share/classes/com/sun/tools/jdi/VirtualMachineImpl.java

```diff
@@ -1445,23 +1445,6 @@ synchronized ObjectReferenceImpl objectMirror(long id, int tag) {
         return object;
     }
 
-    synchronized void removeObjectMirror(ObjectReferenceImpl object) {
-        // Handle any queue elements that are not strongly reachable
-        processQueue();
-
-        SoftObjectReference ref = objectsByID.remove(object.ref());
-        if (ref != null) {
-            batchForDispose(ref);
-        } else {
-            /*
-             * If there's a live ObjectReference about, it better be part
-             * of the cache.
-             */
-            throw new InternalException("ObjectReference " + object.ref() +
-                                        " not found in object cache");
-        }
-    }
-
     synchronized void removeObjectMirror(SoftObjectReference ref) {
```

Comment:
It might able help to make this method removeObjectMirror, this will make it clear that the method can't be called from code in other source files in this package.